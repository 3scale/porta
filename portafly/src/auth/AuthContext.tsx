import React, { createContext, useContext } from 'react'
import { AuthToken } from 'types/auth'
import { KeycloakInstance, KeycloakProfile } from 'keycloak-js'

interface IAuthContext {
  authToken: AuthToken,
  userProfile: KeycloakProfile,
  logout: () => any
}

const AuthContext = createContext<IAuthContext>({
  authToken: null,
  userProfile: {},
  logout: () => {}
})

type Props = {
  keycloak: KeycloakInstance,
  userProfile: KeycloakProfile
}

const AuthProvider: React.FunctionComponent<Props> = ({ keycloak, userProfile, children }) => (
  <AuthContext.Provider value={{
    authToken: keycloak.token as AuthToken,
    logout: keycloak.logout,
    userProfile
  }}
  >
    {children}
  </AuthContext.Provider>
)

const useAuth = () => useContext(AuthContext)

export { AuthProvider, useAuth }
