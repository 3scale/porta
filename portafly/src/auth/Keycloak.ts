import Keycloak from 'keycloak-js'

const keycloakConfig = {
  realm: process.env.REACT_APP_KEYCLOAK_REALM || '3scale',
  url: process.env.REACT_APP_KEYCLOAK_URL,
  'ssl-required': process.env.REACT_APP_KEYCLOAK_SSL,
  clientId: process.env.REACT_APP_KEYCLOAK_CLIENTID || 'portafly',
  'public-client': process.env.REACT_APP_KEYCLOAK_PUBLICK_CLIENT
}

const keycloak = Keycloak(keycloakConfig)
export { keycloak }
