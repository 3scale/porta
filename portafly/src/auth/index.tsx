export * from './AuthContext'
export * from './Keycloak'
