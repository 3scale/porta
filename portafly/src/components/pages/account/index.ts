export { default as AccountPage } from './AccountPage'
export * from './ApplicationsTabContent'
export * from './ApplicationsTabDataListTable'
export * from './OverviewTabContent'
export * from './ProductLink'
