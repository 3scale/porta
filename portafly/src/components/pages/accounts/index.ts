export { default as AccountsIndexPage } from './AccountsIndexPage'
export * from './AccountsTable'
export * from './AccountsDataListTable'
export * from './ActionButtonImpersonate'
export * from './CreateAccountButton'
export * from './ExportAccountsButton'
export * from './AccountOverviewLink'
export * from './utils'
