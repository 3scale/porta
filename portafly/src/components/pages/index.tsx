export * from './accounts'
export { default as Overview } from './Overview'
export * from './applications'
export { default as Login } from './Login'
export * from './plans'
export * from './product'
