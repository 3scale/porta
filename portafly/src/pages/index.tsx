export { default as Overview } from './Overview'
export { default as Applications } from './Applications'
export { default as DeveloperAccounts } from './DeveloperAccounts'
export { default as Login } from './Login'
