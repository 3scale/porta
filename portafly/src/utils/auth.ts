import { AuthToken } from 'types/auth'

const getToken = () => JSON.parse(String(localStorage.getItem('token'))) as AuthToken
const setToken = (token: AuthToken) => (
  token
    ? localStorage.setItem('token', JSON.stringify(token))
    : localStorage.removeItem('token')
)

export { getToken, setToken }
