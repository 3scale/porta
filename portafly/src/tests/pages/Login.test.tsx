import React from 'react'

import { render } from 'tests/custom-render'
import { Login } from 'pages'
import { useAuth } from 'auth'

jest.mock('auth')

it('should render the Login page', () => {
  (useAuth as jest.Mock).mockReturnValue({ authToken: null, setAuthToken: jest.fn() })
  const wrapper = render(<Login />)
  wrapper.debug()

  expect(wrapper.container.firstChild).toMatchSnapshot()
})

it('should not render when there is an auth token already', () => {
  (useAuth as jest.Mock).mockReturnValue({ authToken: 'le token', setAuthToken: jest.fn() })
  const wrapper = render(<Login />)
  wrapper.debug()

  expect(wrapper.container.firstChild).toMatchSnapshot()
})
