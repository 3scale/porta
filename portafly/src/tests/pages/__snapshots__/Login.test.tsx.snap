// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render when there is an auth token already 1`] = `null`;

exports[`should render the Login page 1`] = `
<div
  alt="background_img_alt"
  class="pf-c-background-image css-72krlk"
>
  <svg
    class="pf-c-background-image__filter"
    height="0"
    width="0"
    xmlns="http://www.w3.org/2000/svg"
  >
    <filter
      id="image_overlay"
    >
      <fecolormatrix
        type="matrix"
        values="1 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0"
      />
      <fecomponenttransfer
        color-interpolation-filters="sRGB"
        result="duotone"
      >
        <fefuncr
          tableValues="0.086274509803922 0.43921568627451"
          type="table"
        />
        <fefuncg
          tableValues="0.086274509803922 0.43921568627451"
          type="table"
        />
        <fefuncb
          tableValues="0.086274509803922 0.43921568627451"
          type="table"
        />
        <fefunca
          tableValues="0 1"
          type="table"
        />
      </fecomponenttransfer>
    </filter>
  </svg>
</div>
`;
